vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -0,0 +1,233 @@
+                LocalDateTime.of(date, time) :
+                null;
+    }
+
+    @Override
+    protected void setPresentationValue(
+            LocalDateTime newPresentationValue) {
+        datePicker.setValue(newPresentationValue != null ?
+                newPresentationValue.toLocalDate() :
+                null);
+        timePicker.setValue(newPresentationValue != null ?
+                newPresentationValue.toLocalTime() :
+                null);
+    }
+
+    @Override
+    protected void setModelValue(LocalDateTime newModelValue, boolean fromClient) {
+        super.setModelValue(newModelValue, fromClient);
+    }
+
+    /**
+     * Regenerates the value by calling {@link #generateModelValue()} and updates the model.
+     * If the value is different than the current one, a {@link ValueChangeEvent} will be generated with {@link ValueChangeEvent#isFromClient()} set to <code>true</code>
+     *
+     * <p>This method is called when the web component generates a <b>changed</b> event, typically in response to a change made by the user in one of the contained fields.
+     *
+     * <p>Custom implementations of this method <b>must</b> call {@link #setModelValue(LocalDateTime, boolean)} with the updated model value.
+     * Subclasses can call this method when the model value needs to be regenerated and updated.
+     */
+    protected void updateValue() {
+        // The second parameter is true since this method is called in response to a change event from the client.
+        setModelValue(generateModelValue(), true);
+    }
+
+    /**
+     * Adds the given Component to the specified slot of this component.
+     *
+     * @param component The component to add.
+     * @param slot The name of the slot to add the component to.
+     * @see <a
+     *      href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot">MDN
+     *      page about slots</a>
+     * @see <a
+     *      href="https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element">Spec
+     *      website about slots</a>
+     */
+    private void addToSlot(Component component, String slot) {
+        Objects.requireNonNull(component, "Component to add cannot be null");
+        component.getElement().setAttribute("slot", slot);
+        getElement().appendChild(component.getElement());
+    }
+
+    /**
+     * Adds the given DatePicker to the 'date-picker' slot of this component.
+     *
+     * @param datePicker The DatePicker to add.
+     * @see #addToSlot(Component, String)
+     */
+    private void slotDatePicker(DatePicker datePicker) {
+        addToSlot(datePicker, "date-picker");
+    }
+
+    /**
+     * Adds the given TimePicker to the 'time-picker' slot of this component.
+     *
+     * @param timePicker The TimePicker to add.
+     * @see #addToSlot(Component, String)
+     */
+    private void slotTimePicker(TimePicker timePicker) {
+        addToSlot(timePicker, "time-picker");
+    }
```

Comment:
IMO this is a bit too much. The methods are private and called only in one place, and I don't see any benefit in readability to have
```java
slotDatePicker(datePicker);
```
compared to
```
addToSlot(datePicker, "date-picker");
```

So not worth adding so many lines of code in my opinion.

Reply:
I agree. I'll remove these two extra methods and shorten the comment.